src/aggregation/staging_collector.rs

```diff
@@ -0,0 +1,83 @@
+use super::agg_req_with_accessor::AggregationsWithAccessor;
+use super::intermediate_agg_result::IntermediateAggregationResults;
+use super::segment_agg_result::SegmentAggregationCollector;
+use crate::DocId;
+
+pub(crate) const DOC_BLOCK_SIZE: usize = 64;
+pub(crate) type DocBlock = [DocId; DOC_BLOCK_SIZE];
```

Comment:
Don't we want to box this? That's a lot of data to put on the stack?

Reply:
It's not on the stack because the collector is always boxed